src/rules/no-named-as-default-member.js

```diff
@@ -49,6 +49,8 @@ module.exports = function(context) {
   }
 
   function handleDestructuringAssignment(node) {
+    if (!node.init) return
+
```

Comment:
True. I probably should have put the node.init check after the ObjectPattern test anyway; or something like (node.init && node.init.type === ...) for clarity. 

I generally like to do lots of simple predicates vs. mega-predicates, thus the quick early exit. I'm cool either way, though, feel free to reorder if you'd like.